app/scripts/lib/inpage-provider.js

```diff
@@ -65,9 +56,10 @@ MetamaskInpageProvider.prototype.send = function (payload) {
       result = selectedAddress || '0x0000000000000000000000000000000000000000'
       break
 
-    // fallback to normal rpc
+    // throw not-supported Error
     default:
-      return self.syncProvider.send(payload)
+      var message = 'The MetaMask Web3 object does not support synchronous methods. See https://github.com/MetaMask/faq#all-async---think-of-metamask-as-a-light-client for details.'
```

Comment:
Oh hey I broke this link. It is now:
https://github.com/MetaMask/faq/blob/master/DEVELOPERS.md#all-async---think-of-metamask-as-a-light-client